Neon detection on M1 needs fixing for 1.23.0 release

See https://github.com/numpy/numpy/pull/21705#issuecomment-1154584550

I noticed one failure compiling `cpu_asimd.c` with 1.23.rc3:
```
INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-Werror'
INFO: CCompilerOpt.feature_test[1547] : testing feature 'ASIMD' with flags ()
INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include

INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-Werror'
WARN: CCompilerOpt.dist_test[630] : CCompilerOpt._dist_test_spawn[764] : Command (gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include -Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c /private/scratch.noindex/fink.build/numpy-py310-1.23.0rc3-1/numpy-1.23.0rc3/numpy/distutils/checks/cpu_asimd.c -o /private/tmp/tmpngjq3hpq/private/scratch.noindex/fink.build/numpy-py310-1.23.0rc3-1/numpy-1.23.0rc3/numpy/distutils/checks/cpu_asimd.o -MMD -MF /private/tmp/tmpngjq3hpq/private/scratch.noindex/fink.build/numpy-py310-1.23.0rc3-1/numpy-1.23.0rc3/numpy/distutils/checks/cpu_asimd.o.d -Werror) failed with exit status 1 output -> 
/private/scratch.noindex/fink.build/numpy-py310-1.23.0rc3-1/numpy-1.23.0rc3/numpy/distutils/checks/cpu_asimd.c:17:17: error: incompatible pointer types initializing 'double *' with an expression of type 'float *' [-Werror,-Wincompatible-pointer-types]
        double *src2 = (float*)argv[argc-1];
                ^      ~~~~~~~~~~~~~~~~~~~~
1 error generated.

WARN: CCompilerOpt.feature_test[1563] : testing failed
INFO: CCompilerOpt.feature_test[1547] : testing feature 'NEON' with flags ()
INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include

INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-Werror'
INFO: CCompilerOpt.__init__[1791] : check requested dispatch-able features
INFO: CCompilerOpt.cc_test_flags[1073] : testing flags (-march=armv8.2-a+fp16)
INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include

INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-march=armv8.2-a+fp16'
INFO: CCompilerOpt.feature_test[1547] : testing feature 'ASIMDHP' with flags (-march=armv8.2-a+fp16)
INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include

INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-march=armv8.2-a+fp16 -Werror'
INFO: CCompilerOpt.__init__[1807] : initialize targets groups
INFO: CCompilerOpt.__init__[1809] : parse target group simd_test
INFO: CCompilerOpt._parse_target_tokens[2020] : skip targets (AVX512_SKX SSE2 XOP VSX3 ASIMD VSX2 SSE42 FMA4 VXE VX VSX VSX4 (AVX2 FMA3) AVX512F VXE2 NEON) not part of baseline or dispatch-able features
```
which did not show up prior to #21705
```
INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-Werror'
INFO: CCompilerOpt.feature_test[1522] : testing feature 'ASIMD' with flags ()
INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include

INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-Werror'
INFO: CCompilerOpt.feature_test[1522] : testing feature 'NEON_VFPV4' with flags ()
INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wno-nullability-completeness -Wno-expansion-to-defined -mtune=native -I/opt/include

[numerous other feature_tests]

INFO: compile options: '-Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/opt/include/python3.10 -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/common -Ibuild/src.macosx-12.4-arm64-3.10/numpy/core/src/npymath -c'
extra options: '-march=armv8.2-a+fp16+fp16fml -Werror'
INFO: CCompilerOpt.__init__[1778] : skip features (NEON_FP16 NEON ASIMD NEON_VFPV4) since its part of baseline
INFO: CCompilerOpt.__init__[1782] : initialize targets groups
INFO: CCompilerOpt.__init__[1784] : parse target group simd_test
INFO: CCompilerOpt._parse_target_tokens[1995] : skip targets (SSE2 AVX512F VSX4 VXE2 SSE42 AVX512_SKX FMA4 VSX3 (FMA3 AVX2) VSX2 VSX VXE XOP VX) not part of baseline or dispatch-able features
INFO: CCompilerOpt._parse_policy_not_keepbase[2107] : skip baseline features (ASIMD)
```
Perhaps it just requires a recast of `src2`?

--- a/numpy/distutils/checks/cpu_asimd.c
+++ b/numpy/distutils/checks/cpu_asimd.c
@@ -14,7 +14,7 @@ int main(int argc, char **argv)
     ret += (int)vgetq_lane_f32(vrndq_f32(v1), 0);
 #ifdef __aarch64__
     {
-        double *src2 = (float*)argv[argc-1];
+        double *src2 = (double*)argv[argc-1];
         float64x2_t vd1 = vdupq_n_f64(src2[0]), vd2 = vdupq_n_f64(src2[1]);
         /* MAXMIN */
         ret += (int)vgetq_lane_f64(vmaxnmq_f64(vd1, vd2), 0);

